tests/ship_client.cpp

```diff
@@ -1,65 +1,90 @@
-      ("num-requests,n", bpo::value<uint32_t>(&num_requests)->default_value(num_requests), "number of requests to make");
-   bpo::variables_map varmap;
-   bpo::store(bpo::parse_command_line(argc, argv, cli), varmap);
-   bpo::notify(varmap);
-
-   if(help) {
-      cli.print(std::cout);
-      return 0;
-   }
-
-   std::string::size_type colon = socket_address.find(':');
-   eosio::check(colon != std::string::npos, "Missing ':' seperator in Websocket address and port");
-   std::string statehistory_server = socket_address.substr(0, colon);
-   std::string statehistory_port = socket_address.substr(colon+1);
-
-   std::cerr << "[\n{\n   \"status\": \"construct\",\n   \"time\": " << time(NULL) << "\n},\n";
-
-   try {
-      boost::asio::connect(stream.next_layer(), resolver.resolve(statehistory_server, statehistory_port));
-      stream.handshake(statehistory_server, "/");
+namespace bpo = boost::program_options;
 
+int main(int argc, char *argv[]) {
+  boost::asio::io_context ctx;
+  boost::asio::ip::tcp::resolver resolver(ctx);
+  ws::stream<boost::asio::ip::tcp::socket> tcp_stream(ctx);
+  eosio::abi abi;
+
+  unixs::socket unix_socket(ctx);
+  ws::stream<unixs::socket> unix_stream(ctx);
+
+  bpo::options_description cli("ship_client command line options");
+  bool help = false;
+  std::string socket_address = "127.0.0.1:8080";
+  uint32_t num_requests = 1;
+
+  cli.add_options()("help,h", bpo::bool_switch(&help)->default_value(false),
+                    "Print this help message and exit.")(
+      "socket-address,a",
+      bpo::value<std::string>(&socket_address)->default_value(socket_address),
+      "Websocket address and port.")(
+      "num-requests,n",
+      bpo::value<uint32_t>(&num_requests)->default_value(num_requests),
+      "number of requests to make");
+  bpo::variables_map varmap;
+  bpo::store(bpo::parse_command_line(argc, argv, cli), varmap);
+  bpo::notify(varmap);
+
+  if (help) {
+    cli.print(std::cout);
+    return 0;
+  }
+
+  std::string statehistory_server, statehistory_port;
+
+  // unix socket
+  if (boost::algorithm::starts_with(socket_address, "ws+unix://") ||
+      boost::algorithm::starts_with(socket_address, "unix://")) {
+    statehistory_port = "";
+    statehistory_server = socket_address.substr(socket_address.find("unix://") +
+                                                strlen("unix://") + 1);
+  } else {
+    std::string::size_type colon = socket_address.find(':');
+    eosio::check(colon != std::string::npos,
+                 "Missing ':' seperator in Websocket address and port");
+    statehistory_server = socket_address.substr(0, colon);
+    statehistory_port = socket_address.substr(colon + 1);
+  }
+
+  std::cerr << "[\n{\n   \"status\": \"construct\",\n   \"time\": "
+            << time(NULL) << "\n},\n";
+
+  try {
+    auto run = [&]<typename SocketStream>(SocketStream &stream) {
       {
-         boost::beast::flat_buffer abi_buffer;
```

Comment:
Looks like the reformatter's activities are still present in this file, both here and below.

Reply:
I adjusted clang formatting options and re-formatted with 3 spaces, refs/ptrs on the left, etc. Seems to be close to what we need. Contemts of .clang-format:
`BasedOnStyle: LLVM
UseTab: Never
IndentWidth: 3
PointerAlignment: Left`

Reply:
Clang format seems to have separate knobs for IndentWidth in methods vs functions.  `main()` got reindented to 2 space indent for some reason.  I think something else from the LLVM style needs to be overridden.

I'm not wild about allowing clang format to touch the file because of the long swath in the middle of it that was mis-indented at 5 spaces instead of 6.  Github diff isn't good about highlighting actual code differences as opposed to mere whitespace changes.

Reply:
Re-formatted manually. I will revisit clang-format config later, out of scope of this PR. Main issue is with wrapping long lines and couple more less harmful.